### DISCUSSION TITLE
不再维护 datastore 插件 orm 部分，请迁移至 orm 插件

### DISCUSSION THREAD

**Discussion by he0119** [he0119/nonebot-plugin-datastore#184] [Announcements]
官方的 [orm](https://github.com/nonebot/plugin-orm) 插件已经正式发布，其 orm 功能非常完善。所以 datastore 今后将不在维护 orm 部分。请大家将数据库迁移至官方的 orm 插件。

后面会将迁移方法写在下面的评论部分。

**Comment by he0119** [comment]
以 nonebot-bison 为例 <https://github.com/MountainDash/nonebot-bison/pull/399>：

1. 添加 nonebot-plugin-orm 插件的依赖
2. 删除 migrations 文件夹
3. 用 orm 的 Model 替换 datastore 的，并用 get_session 替换 create_session。
4. 创建新的迁移脚本 `nb orm revision --version-path .\nonebot_bison\config\migrations\ --branch-label nonebot_bison --head base -m "init db"`
5. 配置 orm_version_location
6. 修改测试
7. 迁移 datastore 中的数据

**Reply by AzideCupric** [reply]
按照上述步骤迁移 GuGuMur/nonebot-plugin-skland-arksign#51
第4步的命令可以不需要指定`--version-path`，这会将migrations目录生成在插件的主目录下
生成的migrations目录需要手动新建一个`__init__.py`文件，这样才不会在第六步生成迁移脚本`nb orm revision -m "xxx"`时将新的脚本生成到错误的位置
